pkg/kinflate/app/application_test.go

```diff
@@ -0,0 +1,230 @@
+							"app": "nginx",
+						},
+					},
+					"template": map[string]interface{}{
+						"metadata": map[string]interface{}{
+							"annotations": map[string]interface{}{
+								"note": "This is a test annotation",
+							},
+							"labels": map[string]interface{}{
+								"app": "nginx",
+							},
+						},
+					},
+				},
+			},
+		},
+		types.GroupVersionKindName{
+			GVK:  schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"},
+			Name: "literalConfigMap",
+		}: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "ConfigMap",
+				"metadata": map[string]interface{}{
+					"name": "foo-literalConfigMap-h25f8c59t4",
+					"labels": map[string]interface{}{
+						"app": "nginx",
+					},
+					"annotations": map[string]interface{}{
+						"note": "This is a test annotation",
+					},
+					"creationTimestamp": nil,
+				},
+				"data": map[string]interface{}{
+					"DB_USERNAME": "admin",
+					"DB_PASSWORD": "somepw",
+				},
+			},
+		},
+		types.GroupVersionKindName{
+			GVK:  schema.GroupVersionKind{Version: "v1", Kind: "Secret"},
+			Name: "secret",
+		}: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Secret",
+				"metadata": map[string]interface{}{
+					"name": "foo-secret-2c9kh7fh8t",
+					"labels": map[string]interface{}{
+						"app": "nginx",
+					},
+					"annotations": map[string]interface{}{
+						"note": "This is a test annotation",
+					},
+					"creationTimestamp": nil,
+				},
+				"type": string(corev1.SecretTypeOpaque),
+				"data": map[string]interface{}{
+					"DB_USERNAME": base64.StdEncoding.EncodeToString([]byte("admin")),
+					"DB_PASSWORD": base64.StdEncoding.EncodeToString([]byte("somepw")),
+				},
+			},
+		},
+	}
+	l := setupTest(t)
+	app, err := New(l)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	actual, err := app.Resources()
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	if !reflect.DeepEqual(actual, expected) {
+		err = compareMap(actual, expected)
+		t.Fatalf("unexpected error: %v", err)
+	}
+}
+
+func TestRawResources(t *testing.T) {
+	expected := types.ResourceCollection{
+		types.GroupVersionKindName{
+			GVK:  schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"},
+			Name: "dply1",
+		}: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "apps/v1",
+				"kind":       "Deployment",
+				"metadata": map[string]interface{}{
+					"name": "dply1",
+				},
+			},
+		},
+	}
+	l := setupTest(t)
+	app, err := New(l)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	actual, err := app.RawResources()
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	if !reflect.DeepEqual(actual, expected) {
+		err = compareMap(actual, expected)
+		t.Fatalf("unexpected error: %v", err)
+	}
+}
+
+func compareMap(m1, m2 types.ResourceCollection) error {
```

Comment:
This is can generate more useful error message which can show you which obj if different.

Reply:
Sounds good. I'm suspecting we might have a surprising behavior if you don't use the same function to know if they are the same/find the errors. Can you replace with `if err := compareMap(...); err != nil {` instead of comparing using `DeepEqual`?

Reply:
Done